### PR TITLE
chore: dependency fix

### PR DESCRIPTION
**Pull request by mfw78**
This PR:

1. Fixes the missing dependency issues now by updating the `composable-cow` dependency (upstream moved dependencies).

### CODE REVIEW

**Review by fedgiac — COMMENTED** [review]
Tl;dr: let's merge, see inline commetn for only requested change.

Overall I dislike that these changes are needed: we need to change our contract code because one of our transitive dependencies isn't available anymore.
As we're planning to abandon this version of the CoW AMM I don't think it's worth putting it any work to solve this issue (imho everything needed by the contract should be available inside the repo). The changes are minor enough that I'm ok to merge this without updating the audit.

Note that I got weird errors when trying to run `forge install` from this branch, notably I was being asked for credentials for my GitHub account. I suspect git still tries to fetch the missing repos even if not available on this branch if they are needed for current `main` as well. The only fix I had was cloning the repo from scratch while pointing to this branch.

**Review by fedgiac — APPROVED** [review]
